src/elements/AsteroidBuilder.js

```diff
@@ -0,0 +1,77 @@
+import kbm from "../config/kaboom.js";
+import constants from "../utils/constants.js";
+import resources from "../utils/resources.js";
+import Asteroid from "./Asteroid.js";
+
+class AsteroidBuilder {
+    static getAsteroidCharacterstics() {
+        const randomAsteroidId = Math.ceil(kbm.rand(resources.asteroid.images.length));
+        let randomAsteroidScale = Math.random();
+        const randomXCoord = Math.random() * constants.width;
+
+        if (randomAsteroidScale < 0.4) randomAsteroidScale += 0.5;
+
+        return {
+            randomAsteroidId,
+            randomAsteroidScale,
+            randomXCoord
+        };
+    }
+
+    constructor(nOfAsteroidsAtATime = 1) {
+        this.activeAsteroids = [];
+
+        const timer = setInterval(() => {
+            let i = nOfAsteroidsAtATime;
+            while (i > 0) {
+                let {
```

Comment:
You can declare it outside the loop for better clarity.

Reply:
Every asteroid that is created gets randomised characteristics. Declaring it outside would result in asteroids having identical characteristics.